Doc/library/argparse.rst

```diff
@@ -2162,6 +2162,46 @@ Intermixed parsing
 
    .. versionadded:: 3.7
 
+Registering custom types or actions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. method:: ArgumentParser.register(registry_name, value, object)
+
+   It is possible to register custom actions, types, or other objects with
+   a parser using :meth:`!register`. This method is called when the action is
+   added to the parser. :meth:`!register` takes in the following arguments:
+
+   * *registry_name* - the name of the internal registry where the object
+     will be stored (e.g. ``action``, ``type``).
```

Comment:
Is there a finite set of object types that can be usefully registered in this way?